src/fixed_rational.jl

```diff
@@ -0,0 +1,30 @@
+"""
+    FixedRational{T,den}
+
+A rational number with a fixed denominator. Significantly
+faster than `Rational{T}`, as it never needs to compute
+the `gcd` apart from when printing.
+"""
+struct FixedRational{T<:Integer,den} <: Real
+    num::T
+    global unsafe_fixed_rational(num::Integer, ::Type{T}, ::Val{den}) where {T,den} = new{T,den}(num)
+end
+FixedRational{T,den}(x::Integer) where {T,den} = unsafe_fixed_rational(x * den, T, Val(den))
+FixedRational{T,den}(x::Rational) where {T,den} = unsafe_fixed_rational(widemul(x.num, den) ÷ x.den, T, Val(den))
+
+Base.:*(l::F, r::F) where {T,den,F<:FixedRational{T,den}} = unsafe_fixed_rational(widemul(l.num, r.num) ÷ den, T, Val(den))
+Base.:+(l::F, r::F) where {T,den,F<:FixedRational{T,den}} = unsafe_fixed_rational(l.num + r.num, T, Val(den))
+Base.:-(l::F, r::F) where {T,den,F<:FixedRational{T,den}} = unsafe_fixed_rational(l.num - r.num, T, Val(den))
+Base.:-(x::F) where {T,den,F<:FixedRational{T,den}} = unsafe_fixed_rational(-x.num, T, Val(den))
+Base.inv(x::F) where {T,den,F<:FixedRational{T,den}} = unsafe_fixed_rational(widemul(den, den) ÷ x.num, T, Val(den))
+
+Base.isinteger(x::F) where {T,den,F<:FixedRational{T,den}} = iszero(x.num % den)
+Base.convert(::Type{FixedRational{T,den}}, x::Integer) where {T,den} = unsafe_fixed_rational(x * den, T, Val(den))
+Base.convert(::Type{FixedRational{T,den}}, x::Rational) where {T,den} = FixedRational{T,den}(x)
+Base.convert(::Type{Rational}, x::FixedRational{T,den}) where {T,den} = Rational{T}(x.num, den)
+Base.convert(::Type{AF}, x::FixedRational{T,den}) where {AF<:AbstractFloat,T,den} = convert(AF, x.num / den)
```

Comment:
`x.num//den`

Reply:
It converts to an `AbstractFloat`, so `x.num//den` would be significantly slower as it would have to compute `divgcd` first